File: modelo/src/main/java/sgc/modelo/acomp/Alerta.java
package sgc.modelo.acomp;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.*;
import sgc.modelo.pessoas.Unidade;
import sgc.modelo.base.EntidadeBase;
import sgc.modelo.base.Processo;

import java.time.LocalDateTime;

@Getter
@Entity
@Table(name = "ALERTA")
public class Alerta extends EntidadeBase {
    @ManyToOne
    Unidade unidadeOrigem;

    @ManyToOne
    Unidade unidadeDestino;

    @ManyToOne
    Processo processo;

    String descricao;
    LocalDateTime dataHora;
    boolean lido;
}

File: frontend/src/models/alerta.ts
export interface Alerta {
    codigo: number;
    unidadeOrigem: string;
    unidadeDestino: string;
    dataHora: Date;
    idProcesso: number;
    descricao: string;
}

export interface AlertaServidor {
    codigo: number;
    idAlerta: number;
    idServidor: number;
    lido: boolean;
    dataLeitura: Date | null;
}